Dev main window layout

Briggs-Richardson opened the pull request:
DO NOT MERGE

Briggs-Richardson committed:
Set the layout of the main window

This patch changes the layout of the application to the design described
below.

The layout contains three panels: a video panel, a table panel, and a
button panel. The video panel is in a 16:9 ratio. The button panel is
the same height as the video panel as takes up the remaining window
width. The table panel is below the other two panels, and is the full
width of the window, and takes up the remaining window height. At least
5 rows in the table should be visible at all times.

The video, button, and table panels have been added with this patch.
They are called MediaPanel, CodingAssistancePanel, and TablePanel. The
existing widgets have been moved accordingly.

This patch also adds a placeholder for the button panel, which is
illustrated with 1px black borders and four example buttons.

Testing Steps
  1. Run main.py to start the application.
  2. Verify that the layout of the application matches the description
     above.
  3. Verify the "add row" table button adds a row to the table.
  4. Verify the "add column" table button adds a column to the table.
  5. Verify that a local video file can be loaded and played.

Type: New Feature

Briggs-Richardson committed:
Set the layout of the main window

This patch changes the layout of the application to the design described
below.

The layout contains three panels: a video panel, a table panel, and a
button panel. The video panel is in a 16:9 ratio. The button panel is
the same height as the video panel as takes up the remaining window
width. The table panel is below the other two panels, and is the full
width of the window, and takes up the remaining window height. At least
5 rows in the table should be visible at all times.

The video, button, and table panels have been added with this patch.
They are called MediaPanel, CodingAssistancePanel, and TablePanel. The
existing widgets have been moved accordingly.

Testing Steps
  1. Run main.py to start the application.
  2. Verify that the layout of the application matches the description
     above.
  3. Verify the "add row" table button adds a row to the table.
  4. Verify the "add column" table button adds a column to the table.
  5. Verify that a local video file can be loaded and played.

Type: New Feature

Briggs-Richardson committed:
Set the layout of the main window
This patch changes the layout of the application to the design described
below.

The layout contains three panels: a video panel, a table panel, and a
button panel. The video panel is in a 16:9 ratio. The button panel is
the same height as the video panel as takes up the remaining window
width. The table panel is below the other two panels, and is the full
width of the window, and takes up the remaining window height. At least
5 rows in the table should be visible at all times.

The video, button, and table panels have been added with this patch.
They are called MediaPanel, CodingAssistancePanel, and TablePanel. The
existing widgets have been moved accordingly.

This patch also adds a placeholder for the button panel. The placeholder
is illustrated by a 1px black border, a title, and four example
non-functional coding buttons.

Testing Steps
  1. Run main.py to start the application.
  2. Verify that the layout of the application matches the description
     above.
  3. Verify the "add row" table button adds a row to the table.
  4. Verify the "add column" table button adds a column to the table.
  5. Verify that a local video file can be loaded and played.
  6. Verify that there is a button panel placeholder with four example
     buttons.
  7. Also verify that the encoding assistance panel and button panel
     have 1px black borders and that the button panel has a light gray /
     blue background color.

Type: New Feature

NolanT32 commented on Controllers/controller.py:
```diff
@@ -38,22 +38,23 @@ def __init__(self, window):
 
     @Slot()
     def add_row_to_encoding_table(self):
-        """Add Row Button Logic"""
```
Add a period

Briggs-Richardson replied:
I think this might have been on a different commit. The comment has since been updated to have a period.

NolanT32 commented on View/coding_assistance_panel.py:
```diff
@@ -0,0 +1,45 @@
+        # Create an empty widget to pad the panel.
+        empty_widget = QWidget()
+
+        vertical_layout.addWidget(self._title, stretch=1)
```
Could add comment here for adding the buttons to the vertical layout but may be already implied

Briggs-Richardson replied:
Done.

NolanT32 commented on View/coding_assistance_panel.py:
```diff
@@ -0,0 +1,45 @@
+
+
+class CodingAssistancePanel(QWidget):
+    """Container of all Coding Assistance related widgets."""
```
Are these the buttons to help create user created buttons? Could maybe get a group decision on how we want to call this because I was a little confused with the differentiation between ButtonPanel and CodingAssistancePanel.

Briggs-Richardson replied:
This class is a container for all coding assistance widgets, which for now only includes a title and a button panel to store placeholder coding assistance buttons. The buttons to help users create coding assistance buttons will be added in a future sprint.

NolanT32 commented on View/encoding_table.py:
```diff
@@ -0,0 +1,36 @@
+        Constructor - Sets the properties of a QTableWidget.
+        """
+        super().__init__()
+
+        self.setRowCount(10)
+        self.setColumnCount(4)
+
+        self.horizontalHeader().setStretchLastSection(True)
+        self.horizontalHeader().setSectionResizeMode(QtWidgets.QHeaderView.Stretch)
+        self.verticalHeader().setStretchLastSection(True)
+        self.verticalHeader().setSectionResizeMode(QtWidgets.QHeaderView.Stretch)
```
Could add a comment here about what you are doing

Briggs-Richardson replied:
These changes show as "added" since I renamed the file. The code you're referencing is from Jacob's commit from the last sprint. @Jcobo314 if you have any ideas for a comment here, I'll put them here. Otherwise, it can be changed in another patch.

NolanT32 commented on View/main_window.py:
```diff
@@ -60,20 +51,24 @@ def create_menu_bar(self):
-        """
-        Gets the video widget reference of the View.
+        # Create a container widget for the video and coding assistance panels
+        # to be stored in the first slot in the vertical container layout.
```
Capitilize To

NolanT32 commented on View/table_panel.py:
```diff
@@ -0,0 +1,29 @@
+        super().__init__()
+
+        self.table = EncodingTable()
+        self.add_col_button = QPushButton("Add column")
+        self.add_row_button = QPushButton("Add row")
```
Should column and row be capitalized as it is a name of the button. Up to you to decide

NolanT32 commented on View/button_panel.py:
```diff
@@ -0,0 +1,42 @@
+
+
+class ButtonPanel(QWidget):
+    """Container of all Coding Assistance buttons."""
```
Change this comment as this is the container for all the user created buttons not the buttons to help create them

Briggs-Richardson replied:
User created buttons are referred to as "Coding Assistance buttons". As discussed offline, the buttons to create coding assistance buttons can be "Coding Assistance Creation buttons".

Briggs-Richardson committed:
Set the layout of the main window

This patch changes the layout of the application to the design described
below.

The layout contains three panels: a video panel, a table panel, and a
button panel. The video panel is in a 16:9 ratio. The button panel is
the same height as the video panel as takes up the remaining window
width. The table panel is below the other two panels, and is the full
width of the window, and takes up the remaining window height. At least
5 rows in the table should be visible at all times.

The video, button, and table panels have been added with this patch.
They are called MediaPanel, CodingAssistancePanel, and TablePanel. The
existing widgets have been moved accordingly.

Testing Steps
  1. Run main.py to start the application.
  2. Verify that the layout of the application matches the description
     above.
  3. Verify the "add row" table button adds a row to the table.
  4. Verify the "add column" table button adds a column to the table.
  5. Verify that a local video file can be loaded and played.

Type: New Feature
